### PR TITLE
[dev-menu] update dev-settings to be passed as initial props

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

This is the second PR which passes `DevSettings` as a prop to the dev-menu react app 

# How

This required a bit of refactoring but most of the functionality was already in place, I've just moved the getter from react to native 

# Test Plan

Dev settings should work as expected - test across multiple instances of difference apps and ensure that the settings are working as expected

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuDevSettings.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    return Bundle().apply {
+      putBoolean("isDebuggingRemotely", false)
+      putBoolean("isElementInspectorShown", false)
+      putBoolean("isHotLoadingEnabled", false)
+      putBoolean("isPerfMonitorShown", false)
+    }
+  }
+}
```
Comment (lukmccall) [inline packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuDevSettings.kt:29]:
```suggestion
}

```

File: packages/expo-dev-menu/ios/EXDevMenuDevSettings.m [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      devSettings[@"isElementInspectorShown"] = [NSNumber numberWithBool:bridgeSettings.isElementInspectorShown];
+      devSettings[@"isHotLoadingEnabled"] = [NSNumber numberWithBool:bridgeSettings.isHotLoadingEnabled];
+      devSettings[@"isPerfMonitorShown"] = [NSNumber numberWithBool:bridgeSettings.isPerfMonitorShown];
+    }
+  }
+  
+  return devSettings;
+}
+
+@end
```
Comment (lukmccall) [inline packages/expo-dev-menu/ios/EXDevMenuDevSettings.m:36]: You can leave it for now, but please use Swift for new code if possible.

**Review by lukmccall — APPROVED** [review]

File: packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuAppInfo.kt [hunk 3]
```diff
@@ -6,35 +6,35 @@ import com.facebook.react.bridge.ReactContext
 
 object DevMenuAppInfo {
   fun getAppInfo(reactContext: ReactContext): Bundle {
-    val packageManager = reactContext.packageManager
-    val packageName = reactContext.packageName
-    val packageInfo =  packageManager.getPackageInfo(packageName, 0)
+      val packageManager = reactContext.packageManager
```
Comment (esamelson) [inline packages/expo-dev-menu/android/src/debug/java/expo/modules/devmenu/DevMenuAppInfo.kt:9]: fix indentation in this file (maybe check your IDE indent settings? 🤔 )

**Review by esamelson — APPROVED** [review]
LGTM once tests pass (probably worth re-running the e2e test, looks like a transient failure.)

**Pull request merged**